### PR TITLE
[PERF] Reduce use of cardinality

### PR DESCRIPTION
**Pull request by ashtul**
This PR changes cardinality samples for numeric index from containing all samples to containing 1 in 10 samples. This will save both RAM and CPU.

1. Sample one in 10 numbers for cardinality
2. From forkGC, transfer cardinality stats for existing numbers instead of transferring the cardinality list for deleted numbers. (might consume extra CPU).
3. Cardinality operations move to a `checkCardinality` function so it is visible on performance stats.
4. Initial size of node increased to 16 since it has to be over the sampling quantity which is 10.
5. `NewLeafNode` does not accept `min` and `max` values. Instead, it uses `DBL_MIN` and `DBL_MAX`. Prior to the change, a leaf range was calculated at creation but those numbers were not accurate. Since the MIN and MAX are confirmed with every insert, they aren't required on creation.

On unit tests, an improvement of 50% was obtained.

### CODE REVIEW

File: src/fork_gc.c [hunk 1]
```diff
@@ -414,7 +420,14 @@ static void sendKht(ForkGC *gc, const khash_t(cardvals) * kh) {
     CardinalityValue cu = {.value = u.d48, .appearances = count};
     FGC_SEND_VAR(gc, cu);
     nsent++;
+
+    if (cu.value < minVal) minVal = cu.value;
+    if (cu.value > maxVal) maxVal = cu.value;
```
Comment (ashtul) [inline src/fork_gc.c:425]: might not have min-max in cardinality due to selectiveness. Leave old values.

File: src/fork_gc.c [hunk 2]
```diff
@@ -414,7 +420,14 @@ static void sendKht(ForkGC *gc, const khash_t(cardvals) * kh) {
...
+    uniqueSum += cu.value;
   }
+  FGC_SEND_VAR(gc, minVal);
+  FGC_SEND_VAR(gc, maxVal);
+  FGC_SEND_VAR(gc, uniqueSum);
```
Comment (ashtul) [inline src/fork_gc.c:430]: uniqueSum might not be a good representation.

File: src/numeric_index.c [hunk 3]
```diff
@@ -10,6 +10,7 @@
 #include "util/misc.h"
 //#include "tests/time_sample.h"
 #define NR_EXPONENT 4
+#define NR_CARD_CHECK 10
```
Comment (ashtul) [inline src/numeric_index.c:13]: use in GC

File: src/numeric_index.c [hunk 4]
```diff
@@ -295,7 +304,7 @@ uint16_t numericTreesUniqueId = 0;
 NumericRangeTree *NewNumericRangeTree() {
   NumericRangeTree *ret = rm_malloc(sizeof(NumericRangeTree));
 
-  ret->root = NewLeafNode(2, NF_NEGATIVE_INFINITY, NF_INFINITY, 2);
+  ret->root = NewLeafNode(2, 16);
```
Comment (ashtul) [inline src/numeric_index.c:307]: leave a note about being higher than `NR_CARD_CHECK`

File: src/fork_gc.c (hunk 1 above)
Comment (ashtul) [reply src/fork_gc.c:425]: Removed

File: src/fork_gc.c (hunk 2 above)
Comment (ashtul) [reply src/fork_gc.c:430]: Similar to the current used value.

File: src/numeric_index.c [hunk 5]
```diff
@@ -86,9 +92,9 @@ double NumericRange_Split(NumericRange *n, NumericRangeNode **lp, NumericRangeNo
   double split = (n->unique_sum) / (double)n->card;
```
Comment (MeirShpilraien) [inline src/numeric_index.c:92]: Where do you use this value now?

**Review by MeirShpilraien — APPROVED** [review]
Looks good :+1:

**Review by rafie — APPROVED** [review]

**Pull request merged**